ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron/Aux.hs

```diff
@@ -0,0 +1,545 @@
+    go (CC.MempoolUpdateVote     payload) = recoverBytes payload
+
+-- | Re-encode the mempool payload (without any envelope)
+mempoolPayloadReencode :: CC.AMempoolPayload a -> ByteString
+mempoolPayloadReencode = go
+  where
+    go (CC.MempoolTx             payload) = reencode payload
+    go (CC.MempoolDlg            payload) = reencode payload
+    go (CC.MempoolUpdateProposal payload) = reencode payload
+    go (CC.MempoolUpdateVote     payload) = reencode payload
+
+    reencode :: (Functor f, ToCBOR (f ())) => f a -> ByteString
+    reencode = CBOR.toStrictByteString . toCBOR . void
+
+{-------------------------------------------------------------------------------
+  Annotations
+-------------------------------------------------------------------------------}
+
+reAnnotateMagicId :: ProtocolMagicId -> Annotated ProtocolMagicId ByteString
+reAnnotateMagicId pmi = reAnnotate $ Annotated pmi ()
+
+reAnnotateMagic :: ProtocolMagic -> AProtocolMagic ByteString
+reAnnotateMagic (AProtocolMagic a b) = AProtocolMagic (reAnnotate a) b
+
+reAnnotateBlock :: CC.EpochSlots -> CC.ABlock () -> CC.ABlock ByteString
+reAnnotateBlock epochSlots =
+      (\bs -> splice bs $ CBOR.deserialiseFromBytes
+                            (CC.fromCBORABlock epochSlots)
+                            bs)
+    . CBOR.toLazyByteString
+    . CC.toCBORBlock epochSlots
+  where
+    splice :: Show err
+           => Lazy.ByteString
+           -> Either err (Lazy.ByteString, CC.ABlock ByteSpan)
+           -> CC.ABlock ByteString
+    splice _ (Left err) =
+      error $ "annotateBlock: serialization roundtrip failure: " <> show err
+    splice bs (Right (_leftover, txAux)) =
+      (Lazy.toStrict . slice bs) <$> txAux
+
+reAnnotateBoundary :: ProtocolMagicId
+                   -> CC.ABoundaryBlock ()
+                   -> CC.ABoundaryBlock ByteString
+reAnnotateBoundary pm =
+      (\bs -> splice bs (CBOR.deserialiseFromBytes
+                           CC.fromCBORABoundaryBlock
+                           bs))
+    . CBOR.toLazyByteString
+    . CC.toCBORABoundaryBlock pm
+  where
+    splice :: Show err
+           => Lazy.ByteString
+           -> Either err (Lazy.ByteString, CC.ABoundaryBlock ByteSpan)
+           -> CC.ABoundaryBlock ByteString
+    splice _ (Left err) =
+      error $ "annotateBoundary: serialization roundtrip failure: " <> show err
+    splice bs (Right (_leftover, boundary)) =
+      (Lazy.toStrict . slice bs) <$> boundary
+
+{-------------------------------------------------------------------------------
+  Header of a regular block or EBB
+
+  The ledger layer defines 'ABlockOrBoundary', but no equivalent for headers.
+-------------------------------------------------------------------------------}
+
+data ABlockOrBoundaryHdr a =
+    ABOBBlockHdr    !(CC.AHeader         a)
+  | ABOBBoundaryHdr !(CC.ABoundaryHeader a)
+  deriving (Eq, Show, Functor, Generic, NoUnexpectedThunks)
+
+fromCBORABlockOrBoundaryHdr :: CC.EpochSlots
```

Comment:
Comment: Verify and document what (if anything) this should be binary compatible with.

Reply:
Since this type doesn't exist in `cardano-sl`, we can basically pick whatever we want. This PR does not change the format. It must be compatible with `encodeByronHeader`, but that already has a comment (that it must be compatible with `fromCBORABlockOrBoundaryHdr`).